src/Compiler/Checking/TailCallChecks.fs

```diff
@@ -56,6 +56,10 @@ type TailCall =
         | TailCall.Yes _ -> TailCall.Yes TailCallReturnType.NonVoid
         | TailCall.No -> TailCall.No
 
+let hasTailCallAttrib (attribs: Attribs) =
+    attribs
+    |> List.exists (fun a -> a.TyconRef.CompiledRepresentationForNamedType.FullName = "Microsoft.FSharp.Core.TailCallAttribute")
+
```

Comment:
This will probably need to be in TcGlobals as other attributes and have usual checks as well (tryFindAttribute and co). 

Like we do with IsReadOnlyAttribute (for required attributes as well iirc) for example. You can even do exactly like that, and compiler will generate TailCallAttribute if it doesn't find it in ccus.